packages/url/src/prepend-http.js

```diff
@@ -24,7 +24,7 @@ export function prependHTTP( url ) {
 
 	url = url.trim();
 	if ( ! USABLE_HREF_REGEXP.test( url ) && ! isEmail( url ) ) {
-		return 'http://' + url;
+		return 'https://' + url;
```

Comment:
Actually, I'd generally be against that change, because I'd expect a function called `prependHTTP` to prepend `http`, not `https`. I'd suggest one of these alternative approaches instead:

1. Create a new universal function `prependProtocol` that will look at the current protocol (`window.location.protocol` if in a browser environment), and consider it when deciding whether to apply `http` or `https`. That way we'll keep using `http` for local installations that are not on HTTPS, but we'll still properly support `https` when applicable.
2. Create a function `prependHTTPS` that always prepends `https` and in another PR, use that function instead of `prependHTTP` to make it transparent as to what we're doing. That will also help us test the impact of that change in all areas where `prependHTTP` is being used.

In any case, we might want to deprecate `prependHTTP` as it will no longer be used and preferred, since sites are recommended to be on `https`. 

What do you think?

Reply:
Yeh so I went with adding an option to the existing function in https://github.com/WordPress/gutenberg/pull/47312.

But we could just as easily create a new function and deprecate this one.

Reply:
I agree with you and I think that adding an extra param is of the smallest impact here, since it's not a breaking change. That's why I'm happy to add it here. The downside is the potential confusion that the `prependHTTP` function can prepend `https` but I guess if we document it well, it can be fine.